### PR TITLE
Improved tooltip removal

### PR DESCRIPTION
**Pull request by eraderna**
- Removes hover rules and associated `:transform`s completely, leaving a displayable plot.
- Disables other tooltips.
- Can be applied safely to any plot, not just the `line-and-ribbon-and-rule-plot` ones with the hover rule tooltip, so can use as a filter for all plots going through to kixi.transcode.

### CODE REVIEW

File: src/witan/send/adroddiad/vega_specs/lines.clj [hunk 1]
```diff
@@ -327,18 +330,22 @@
...
+                                                  (some->> m :params (some #(some->> % :name (= "hover"))))))))
+    ;; Remove any `:transform` layers within layers
+    (update $ :layer (partial mapv (fn [m] (if (contains? m :layer)
+                                             (update m :layer (partial remove #(contains? % :transform)))
+                                             m))))
+    ;; Disable any remaining `:tooltip`s by setting value to `nil`
+    (walk/prewalk (fn [e] (if (and (map-entry? e) (= (key e) :tooltip))
+                            [(key e) nil]
+                            e))
+                  $)))
```
Comment (otfrom) [inline src/witan/send/adroddiad/vega_specs/lines.clj:351]: shiny

File: src/witan/send/adroddiad/vega_specs/lines.clj [hunk 2]
```diff
@@ -106,13 +107,15 @@
                                             :size        150
                                             :strokewidth 0.5}}}
                             {:mark     "errorband"
-                             :encoding {:y     {:field oru :title y-title :type "quantitative"}
-                                        :y2    {:field orl}
-                                        :color {:field group :title group-title}}}
+                             :encoding {:y       {:field oru :title y-title :type "quantitative"}
+                                        :y2      {:field orl}
+                                        :color   {:field group :title group-title}
+                                        :tooltip nil}}
```
Comment (otfrom) [inline src/witan/send/adroddiad/vega_specs/lines.clj:113]: does this need `:tooltip nil`?

File: src/witan/send/adroddiad/vega_specs/lines.clj [hunk 3]
```diff
@@ -106,13 +107,15 @@
...
                             {:mark     "errorband"
-                             :encoding {:y     {:field iru :title y-title :type "quantitative"}
-                                        :y2    {:field irl}
-                                        :color {:field group :title group-title}}}
+                             :encoding {:y       {:field iru :title y-title :type "quantitative"}
+                                        :y2      {:field irl}
+                                        :color   {:field group :title group-title}
+                                        :tooltip nil}}
```
Comment (otfrom) [inline src/witan/send/adroddiad/vega_specs/lines.clj:118]: and again

**Review by otfrom — APPROVED** [review]
not sure about the nils in the maps, but :ship:

File: src/witan/send/adroddiad/vega_specs/lines.clj (hunk 2 above)
Comment (eraderna) [reply src/witan/send/adroddiad/vega_specs/lines.clj:113]: Not needed for successful transcode, but without them you are left with the default tooltips after removing the hover rule. So put these in to actually get a plot with no tooltips (and so "remove-tooltips" actually does what it says).

**Review by seb231 — APPROVED** [review]

**Pull request merged**